src/main/java/seedu/address/logic/commands/FilterTaskCommand.java

```diff
@@ -0,0 +1,178 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CATEGORY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.model.Model;
+import seedu.address.model.task.TaskCategory;
+import seedu.address.model.task.TaskCategoryAndDeadlinePredicate;
+import seedu.address.model.task.TaskContainsCategoryPredicate;
+import seedu.address.model.task.TaskDate;
+import seedu.address.model.task.TaskDeadlineBeforeDatePredicate;
+
+/**
+ * Filters and lists all tasks in address book with category that matches the argument category
+ * and / or with deadline that matches the argument date.
+ */
+public class FilterTaskCommand extends Command {
+
+    public static final String COMMAND_WORD = "filter";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Filters all tasks based on the user input "
+            + "and displays them as a list with index numbers.\n"
+            + "Parameters: "
+            + "[" + PREFIX_CATEGORY + "CATEGORY (database/frontend/backend/uiux/presentation/others)] "
+            + "[" + PREFIX_DATE + "DATE (YYYY-MM-DD)]\n"
+            + "Requirement: At least one parameter to filter with must be provided.\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_CATEGORY + "backend "
+            + PREFIX_DATE + "2022-12-12";
+
+    private final FilterTaskDescriptor filterTaskDescriptor;
+    private final TaskCategoryAndDeadlinePredicate both;
+    private final TaskContainsCategoryPredicate category;
+    private final TaskDeadlineBeforeDatePredicate date;
```

Comment:
Is it possible to rename the predicates. Maybe datePredicate/ categoryPredicate or any other clearer names.

Reply:
Resolved and renamed with the updated `TaskCategoryAndDeadlinePredicate` class!